### DISCUSSION TITLE
PrismaClientValidationError:

### DISCUSSION THREAD

**Discussion by rajanshresth** [prisma/prisma#19505] [Q&A]
### Bug description

This is my prisma schema file
![image](https://github.com/prisma/prisma/assets/76089000/ca75ea05-0024-4d84-bbc1-566d14f71d5b)

And, I've created api end point for POST
![image](https://github.com/prisma/prisma/assets/76089000/dbb9a959-b440-4ad0-b8b6-d43105f90d96)

I expect it to be working, instead i get:
![image](https://github.com/prisma/prisma/assets/76089000/89bf0e07-643c-4828-8aef-0143ff831620)

### How to reproduce

Go to: https://github.com/rajanshresth/rkpost.git

### Expected behavior

Create a json content when api endpoint is send as:
![image](https://github.com/prisma/prisma/assets/76089000/460413b5-361f-4d92-9511-8400f4bed9d6)

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url = env("DATABASE_URL")
  relationMode = "prisma"
}

model Post{
  id Int @id @default(autoincrement())
  title String
  content String
  published Boolean @default(false)
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```

```ts
import { PrismaClient } from "@prisma/client";
const prisma = new PrismaClient();
export default prisma;

async function main() {
  const getPost = await prisma.post.findMany();
  const createPost = async (title: string, content: string) => {
    const res = await prisma.post.create({
      data: {
        title: title,
        content: content,
      },
    });
    return res;
  };
  return { getPost, createPost };
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });

```

### Environment & setup

- OS: Ubuntu 22.04.2 LTS
- Database:PlanetScale,
- Node.js version: 20.2.0

### Prisma Version

```
prisma version: 4.14.1

```

**Comment by ludralph** [comment]
Hi @rajanshresth 👋 

I'm able to reproduce your error and i can deduce that you are not correctly passing the fields `title` and `content` to the Prisma `create` function.

i modified your code to deduce this like so in the api endpoint for Post:
```js
const { title, content } =
    typeof req.body == "string" ? JSON.parse(req.body) : req.body;
    if (!title || !content) {
      return NextResponse.json({ error: "Missing required fields: title or content" });
    }
```
this is the output
<img width="1001" alt="image" src="https://github.com/prisma/prisma/assets/13855608/0778b7a3-ffd2-487e-963a-b37915a716d2">

You will need to ensure req.body is correctly returning the fields for `title` and `content`

**Reply by ludralph** [reply]
Hey @rajanshresth,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
